reference/extensions/python_requires.rst

```diff
@@ -1,4 +1,304 @@
 .. _reference_extensions_python_requires:
 
 Python requires
----------------
\ No newline at end of file
+===============
+
+
+Introduction
+------------
+
+The ``python_requires`` feature is a very convenient way to share files and code between
+different recipes. A python requires is a special recipe that do not create packages and
+it is just intended to be reused by other recipes.
+
+A very simple recipe that we want to reuse could be:
+
+.. code-block:: python
+    
+    from conan import ConanFile
+
+    myvar = 123
```

Comment:
that's the scope/life cycle of this `global` (module) variable?
e.g. if one recipe does:
```
self.python_requires["pyreq"].module.myvar = 777
```
will other recipe see this change?

Reply:
Yes